### PR TITLE
Lose the index before 'na_value' assignment in 'to_numpy'

### LINKED ISSUE #45774: BUG: Multi-indexed Series to Numpy fails when `na_value` supplied
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import numpy as np
import pandas as pd

index = pd.MultiIndex.from_tuples([(0, "a"), (0, "b")])
x = pd.Series([1, 2], index=index)

print(x.to_numpy(dtype="float")) # [1. 2.]
print(x.to_numpy(dtype="float", na_value=np.nan)) # KeyError: 1
```

### Issue Description

When converting a multiindexed series to a numpy array, it works correctly when no `na_value` is specified. But when an `na_value` is specified, even if the series has no NA values, it errors.

Digging into this issue, when `na_value` is specified, a branch is taken which sets the `na_value` in the numpy array.
`result[self.isna()] = na_value`

However, the numpy `result` array cannot understand the multiindex of the `self.isna()` multi-indexed array.

The Pull Request https://github.com/pandas-dev/pandas/pull/45775 is one possibility to address this issue.

### Expected Behavior

```python
import numpy as np
import pandas as pd

index = pd.MultiIndex.from_tuples([(0, "a"), (0, "b")])
x = pd.Series([1, 2], index=index)

print(x.to_numpy(dtype="float")) # [1. 2.]
print(x.to_numpy(dtype="float", na_value=np.nan)) # [1. 2.]
```

### Installed Versions

The `pd.show_versions()` produces an error, but below are the versions of the relevant packages.
<details>
python                    3.10.1          h1248fe1_2_cpython    conda-forge

pandas                    1.5.0.dev0+268.gbe8d1ec880           dev_0    <develop>  # Installed locally
numpy                     1.22.0                   pypi_0    pypi
</details>

### PR DESCRIPTION
**Pull request by DamianBarabonkovQC**
- [X] closes #45774 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

This Pull Request addresses the issue where casting a multi-indexed Series `to_numpy` fails if an `na_value` is supplied. See the issue #45774 for more information.

The choice to use `asanyarray` over other solutions to lose the index is one based on performance and simplicity. The following simple benchmark was used to make this determination:

```python
import time
import pandas as pd
import numpy as np

t0 = time.time()

series = pd.Series(range(100000000))

#res = series.isna() # Orignal buggy: Elapsed time:  0.19312691688537598 sec
#res = tuple(series.isna()) # Elapsed time:  5.303609132766724 sec
#res = series.isna().reset_index(drop=True) # Elapsed time:  0.2203369140625 sec
#res = series.isna().to_numpy()  # Elapsed time:  0.19327998161315918 sec
#res = np.array(series.isna())  # Elapsed time:  0.25583529472351074 sec
res = np.asanyarray(series.isna())  # Elapsed time:  0.19619202613830566 sec

t1 = time.time()

#print(res)
print("Elapsed time: ", t1 - t0)
```

This demonstrates that there is no significant overhead to using the `to_numpy` operation.

### CODE REVIEW

**Review by phofl — CHANGES_REQUESTED** [review]
Please add tests. Also this broke existing tests

File: doc/source/whatsnew/v1.5.0.rst [hunk 1]
```diff
@@ -286,7 +286,7 @@ Missing
 
 MultiIndex
 ^^^^^^^^^^
--
+- Bug in :meth:`IndexOpsMixin.to_numpy` where multiindexed Series could not be converted to numpy arrays when an ``na_value`` was supplied (:issue:`45774`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.5.0.rst:289]: the note would need to be user facing

**Review by jreback — CHANGES_REQUESTED** [review]
tests pls

File: doc/source/whatsnew/v1.5.0.rst (hunk 1 above)
Comment (DamianBarabonkovQC) [reply doc/source/whatsnew/v1.5.0.rst:289]: What do you mean by this comment? I do not understand completely what "needs to be user facing" means. Thanks.

Comment (DamianBarabonkovQC) [reply doc/source/whatsnew/v1.5.0.rst:289]: Hey @jreback . May you please elaborate on your comment.

Comment (phofl) [reply doc/source/whatsnew/v1.5.0.rst:332]: ``IndexOpsMixin`` is private, this is not exposed through the api

Comment (DamianBarabonkovQC) [reply doc/source/whatsnew/v1.5.0.rst]: Comment should be addressed.

**Review by jreback — CHANGES_REQUESTED** [review]
pls merge master as well

File: pandas/tests/base/test_conversion.py [hunk 2]
```diff
@@ -415,6 +415,42 @@ def test_to_numpy_na_value_numpy_dtype(
     tm.assert_numpy_array_equal(result, expected)
 
 
+@pytest.mark.parametrize(
```
[the remark this answers is not in the document]
Comment (DamianBarabonkovQC) [reply pandas/tests/base/test_conversion.py:418]: Included

File: pandas/core/base.py [hunk 3]
```diff
@@ -541,7 +541,9 @@ def to_numpy(
         if copy or na_value is not lib.no_default:
             result = result.copy()
             if na_value is not lib.no_default:
-                result[self.isna()] = na_value
+                # Convert the mask to numpy in order to prevent possible
```
[the remark this answers is not in the document]
Comment (DamianBarabonkovQC) [reply pandas/core/base.py:544]: Removed.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @DamianBarabonkovQC

**Pull request merged**